File: 23-03-2024/Fibonacci series up to Nth term.js
//{ Driver Code Starts

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split("\n").map(string => {
        return string.trim();
    });

    main();
});

function readLine() {
    return inputString[currentLine++];
}


function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++){
        
        let n = parseInt(readLine());
        
        let obj = new Solution();
        let res = obj.Series(n);
        
        let S_res = '';
        for(let i=0;i<res.length;i++)
        {
            S_res+=(res[i]);
            S_res+=' ';
        }
        console.log(S_res);
        
    }
}

// } Driver Code Ends





class Solution {
/**
* @param number n

* @returns number[]
*/
    Series(n) {
//Your Code Start
         const MOD = 1e9 + 7;
        let fib = [0, 1];

        for (let i = 2; i <= n; i++) {
            fib[i] = (fib[i - 1] + fib[i - 2]) % MOD;
        }

        return fib.slice(0, n + 1);
        //Your Code End
    }
}
        

File: 23-03-2024/Fibonacci series up to Nth term.py
#User function Template for python3

class Solution:
    def series(self, n):
        # Your Code Start
        val = [0,1]
        for i in range(2,n+1):
            x = val[-1]+val[-2]
            val.append(x%(10**9+7))
        return val
    #Your Code End



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        result = ob.series(N)
        print(*result)
# } Driver Code Ends
